### PR TITLE
feat(recipe): add GLM-5.2 753B pretrain base recipe

### PR DESCRIPTION
**Pull request by nvegesna-netizen**
## Summary

Adds a **NullTokenizer mock-data pre-training recipe** for GLM-5.2 753B MoE, following the convention established by every other large MoE family in this repo (deepseek v2/v3/v4, gpt_oss, nemotronh, moonlight, olmoe, kimi k2 — all use NullTokenizer for their pretrain base recipe). The existing `glm52_pretrain_416gpu_h100_bf16_config` is a bounded 100-step real-data run suited for CI validation; this recipe fills the complementary slot.

`glm52_753b_pretrain_512gpu_h100_bf16_config()` appends to `glm5.py` alongside the existing four H100 recipes. It is built from the official HF config through `GLM5Bridge` / `AutoBridge.from_hf_pretrained("zai-org/GLM-5.2")` and follows the standard conventions (`_pretrain_common()`, NullTokenizer, `CommOverlapConfig`, `COMMON_RECIPE_ENV_VARS`).

- DSA-safe defaults: CP=1, RoPE fusion disabled.
- MTP enabled with one prediction layer, loss weight 0.1, explicit placement before loss in the PP layout.
- Recommended starting parallelism TP=2, PP=16, EP=32 on 512 H100 GPUs; selective `mla_up_proj` recompute.
- `cuda_graph_scope=None` (consistent with `cuda_graph_impl="none"`).
- Exported through `recipes/glm/h100/__init__.py`; unit test uses an isolated `_FakeBridgeNullTok` mock to avoid touching the existing test fixtures.

### CODE REVIEW

File: src/megatron/bridge/recipes/glm/glm5.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2026, NVIDIA CORPORATION.  All rights reserved.
```
Comment (yaoyu-33) [inline src/megatron/bridge/recipes/glm/glm5.py:1]: we restructured things - this needs to be put inside H100 folder now or if it's GB then GB folder.

Comment (nvegesna-netizen) [reply src/megatron/bridge/recipes/glm/glm5.py:1]: Restructured.

**Review by yaoyu-33 — DISMISSED** [review]
